Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp

```diff
@@ -799,6 +802,26 @@ WebKitWebViewAccessible* View::accessible() const
 }
 #endif
 
+#if ENABLE(WPE_PLATFORM)
+void View::handleGesture(WPEGesture gesture, WPEGestureController& controller)
+{
+    // Some gestures require immediate feedback, so they are detected and handled right away.
+    switch (gesture) {
+    case WPE_GESTURE_NONE:
+    case WPE_GESTURE_TAP:
+        break;
+    case WPE_GESTURE_DRAG:
+        if (double x, y, dx, dy; wpe_gesture_controller_get_detected_gesture_position(&controller, &x, &y) && wpe_gesture_controller_get_detected_gesture_delta(&controller, &dx, &dy)) {
+            auto* simulatedScrollEvent = wpe_event_scroll_new(
+                m_wpeView.get(), WPE_INPUT_SOURCE_MOUSE, 0, static_cast<WPEModifiers>(0), dx, dy, TRUE, FALSE, x, y
+            );
```

Comment:
We need to detect somehow the beginning of the drag to make sure that the generated event uses `WebWheelEvent::Phase::Began`.

Reply:
Use GRefPtr and change the source to touchscreen.